test/test-clusters/src/main/java/org/elasticsearch/test/cluster/local/LocalClusterFactory.java

```diff
@@ -0,0 +1,485 @@
+                    try {
+                        ProcessUtils.exec(
+                            workingDir,
+                            distributionDir.resolve("bin").resolve("elasticsearch-users"),
+                            getEnvironmentVariables(),
+                            false,
+                            "useradd",
+                            user.getUsername(),
+                            "-p",
+                            user.getPassword(),
+                            "-r",
+                            user.getRole()
+                        ).waitFor();
+                    } catch (InterruptedException e) {
+                        throw new RuntimeException(e);
+                    }
+                }
+            }
+        }
+
+        private void installPlugins() {
+            if (spec.getPlugins().isEmpty() == false) {
+                Pattern pattern = Pattern.compile("(.+)(?:-\\d\\.\\d\\.\\d-SNAPSHOT\\.zip)?");
+
+                LOGGER.info("Installing plugins {} into node '{}", spec.getPlugins(), spec.getName());
+                List<Path> pluginPaths = Arrays.stream(System.getProperty(TESTS_CLUSTER_PLUGINS_PATH_SYSPROP).split(File.pathSeparator))
+                    .map(Path::of)
+                    .toList();
+
+                List<String> toInstall = spec.getPlugins()
+                    .stream()
+                    .map(
+                        pluginName -> pluginPaths.stream()
+                            .map(path -> Pair.of(pattern.matcher(path.getFileName().toString()), path))
+                            .filter(pair -> pair.left.matches())
+                            .map(p -> p.right.getParent().resolve(p.left.group(1)))
+                            .findFirst()
+                            .orElseThrow(() -> {
+                                String taskPath = System.getProperty("tests.task");
+                                String project = taskPath.substring(0, taskPath.lastIndexOf(':'));
+
+                                throw new RuntimeException(
+                                    "Unable to locate plugin '"
+                                        + pluginName
+                                        + "'. Ensure you've added the following to the build script for project '"
+                                        + project
+                                        + "':\n\n"
+                                        + "dependencies {\n"
+                                        + "  clusterModules "
+                                        + "project(':plugins:"
+                                        + pluginName
+                                        + "')"
+                                        + "\n}"
+                                );
+                            })
+                    )
+                    .map(p -> p.toUri().toString())
+                    .toList();
+
+                if (spec.getVersion().onOrAfter("7.6.0")) {
```

Comment:
Plugins are not installable before 7.6? Don't we need an else to loop through each plugin vs doing the batch that we do now?

Reply:
My bad. The difference is that before 7.6.0 the CLI doesn't support installing multiple plugins in one execution. I just forgot to implement the `else` here 🤦. I've fixed this.